kong/api/routes/config.lua

```diff
@@ -20,6 +20,29 @@ local _reports = {
 
 return {
   ["/config"] = {
+    GET = function(self, db)
+      if kong.db.strategy ~= "off" then
+        return kong.response.exit(400, {
+          message = "this endpoint is only available when Kong is " ..
+                    "configured to not use a database"
+        })
+      end
+
+      local file = {
+        buffer = {},
+        write = function(self, str)
+          self.buffer[#self.buffer + 1] = str
+        end,
+      }
+
+      local ok, err = declarative.export_from_db(file)
+      if not ok then
+        kong.log.err("failed exporting config from cache: ", err)
+        return kong.response.exit(500, { message = "An unexpected error occurred" })
```

Comment:
A while ago we discussed not using this, but instead throwing an error. Or did I misunderstand that?

Reply:
afaik throwing error will call kong_error_handlers which is very different than response.exit. that should be fixed first I think.

Reply:
Also with throwing errors, you need to be very careful where you do so. E.g. if you do it on `header_filter` it will just cause Nginx to close the connection immediately, and that is not what we want. We should revisit this as a whole at some point.

Reply:
Afaik throwing an error results in a 500, and always a proper table error as above (we've seen occurrences where the manual errors were just strings, so we'd prevent that), and also a stack trace in the logs (so also no need for the extra manual log line just above).

@bungle why wouldn't you want to close the connection immediately in the header_filter phase? If an error occurs, then the request is in an undetermined state, so imo nothing else you can do, but close the connection.

Reply:
@Tieske currently we catch errors in plugins on some phases and I think with admin api, but nit all phases, like header_filter. I think it would be better to give a proper 500 error on header_filter too rather than just closing connection and responding with nothing.

Reply:
Yes, a proper 500 is a requirement.

That said; why the inconsistency in error handling?

Reply:
@Tieske originally we didn't handle errors in any phase, but when we added delayed response, we wrote it with `coroutine:wrap` that is a `pcall` with steroids. I am not sure if we want to be 100% consistent. E.g. handling error in body_filter is tricky, we cannot change headers anymore, and perhaps some content was already sent. And in log phase, we should perhaps continue with next plugin even if some other plugin fails. But... well this is getting off-topic, sorry for hijacking.

Reply:
this is tracked elsewhere, so resolving this.